src/coreclr/src/zap/zapinfo.cpp

```diff
@@ -2182,7 +2192,20 @@ DWORD FilterNamedIntrinsicMethodAttribs(ZapInfo* pZapInfo, DWORD attribs, CORINF
                     fTreatAsRegularMethodCall = !fIsGetIsSupportedMethod;
                 }
             }
+#endif // defined(TARGET_X86) || defined(TARGET_AMD64)
+#ifdef TARGET_X86
```

Comment:
I wonder if it is wroth minimally recognizing the `IsSupported` methods as "intrinsic" on ARM32 to improve the codegen there as well...

Reply:
What's the HWIntrinsics support state on ARM32 now?  Do you mean just treating `Arm64` as intrinsic so it can be eliminated, or should we be keeping `ArmBase` intrinsic unconditionally for ARM32 as well?

Reply:
ARM32 currently has no HWIntrinsic or SIMDIntrinsic support and it won't have it in .NET 5. I don't think it even recognizes the `IsSupported` checks as intrinsic and so I'm unsure if they are being properly "removed" during crossgen.

Reply:
Ah, ok.  On ARM32, the `PlatformNotSupported` variants are included for all HWIntrinsics classes, so `IsSupported` will be a `false` constant for every ISA:

https://github.com/dotnet/runtime/blob/16c7b3148ab9892b3155295f7c24afb7cf67e745/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems#L1802-L1809

I'd rather not add code that doesn't do anything for now.  This function will need to be revisited if/when ARM32 gets HWIntrinsincs supports anyway.